### PR TITLE
Follow author UX improvement #465 

### PR DESCRIPTION
**Pull request by dgnaegi**
idea / concept:
Initialize subscribed authors at first page rendering / "load more"-button-click and use the initialized data as sorting criteria

behavior

### CODE REVIEW

File: components/Notifications/SubscribedAuthors.js [hunk 1]
```diff
@@ -153,3 +176,20 @@ const SubscribedAuthors = ({
 }
 
 export default compose(withT, graphql(myUserSubscriptions))(SubscribedAuthors)
+
+function InitializeSubscribedAuthorIds(
```
Comment (annatraussnig) [inline components/Notifications/SubscribedAuthors.js:180]: The function name I'd start with lowercase. To be more consistent with the rest of the codebase, I'd also bring it inside the component and use `setInitialySubscribedAuthorIds`directly. That way, the `export default`also remains the last line of the file, which is also consistent with our codebase.

File: components/Notifications/SubscribedAuthors.js [hunk 2]
```diff
@@ -65,6 +65,19 @@ const SubscribedAuthors = ({
...
+    setInitiallySubscribedAuthorIds
+  ] = useState([])
+
+  useEffect(() => {
+    InitializeSubscribedAuthorIds(
+      authors,
+      myUserSubscriptions.subscribedTo.nodes,
+      setInitiallySubscribedAuthorIds
+    )
+  }, [])
```
Comment (annatraussnig) [inline components/Notifications/SubscribedAuthors.js:79]: Don't we run into the risk that `authors` and `myUserSubscriptions`are still undefined at this point?

**Review by annatraussnig — COMMENTED** [review]
Hey, first thank you so much again! I just tried to run the code locally and faced 2 issues:

1. undefined authors error upon first loading the user preferences
2. the sorting in the expanded list is upside down (followed authors at the bottom -> I made a screenshot)

Screenshot 2020-10-19 at 16 26 13

If you have to time to have a look, that would be very cool indeed. If not, let me know so I'm aware and I can take over.

**Comment by dgnaegi** [comment]
> If you have to time to have a look, that would be very cool indeed. If not, let me know so I'm aware and I can take over.

Thanks for the review :) Regarding the issues:

1) I didnt noticed this one :-/ Under the assumption that the list is intitially collapsed I could remove the useeffect completely so the list gets initiallized after expanding. But this could make further adjustments more complex. Would this be a solution? I already pushed the change but didnt test it yet. Will do so in the next days

2) On my machine, the sorting is correctly:

grafik

Could it be an enviroment issue? I noticed that on your screenshot there are some magazine-checkboxes missing as well compared to the production environment. Why is that?

On a logical side it is: Included in the list -> true -> 1 -> In descending order on top of 0

But i could dig into this issue as well, probably on thursday

**Comment by annatraussnig** [comment]
Re 2. yes, those profile are still from Wahltindär. Are you connecting to the live DB or what's your setup? If you want to try and follow some the politicians and see if this plays a role, here are a couple of randomly chosen links:
https://www.republik.ch/~rsauter
https://www.republik.ch/~joergmaeder
https://www.republik.ch/~bwalti
I'm curious if this changes the game!

**Comment by dgnaegi** [comment]
> Re 2. yes, those profile are still from Wahltindär. Are you connecting to the live DB or what's your setup? If you want to try and follow some the politicians and see if this plays a role, here are a couple of randomly chosen links:
> https://www.republik.ch/~rsauter
> https://www.republik.ch/~joergmaeder
> https://www.republik.ch/~bwalti
> I'm curious if this changes the game!

Please excuse the late reply, I was pretty busy the last couple of days. Thanks for the hint with the wahltindär, but this doesn't change the behavior on my machine. I do work with the Production-API's

grafik

You don't have to invest to much time into my PR, but im still wondering why we experience a different behavior...

**Review by annatraussnig — APPROVED** [review]

**Comment by annatraussnig** [comment]
Hey Daniel! Thanks for your patience 😸 I also got busy with Urabstimmung work... Anyway thanks for the fixes, all seems to work now! I can merge the PR on Friday morning. It's a bit late for hacktoberfest… although you did most of the work in October… Shall I add the tag nonetheless or doesn't matter anymore?

**Pull request merged**